payments/models.py

```diff
@@ -133,6 +133,68 @@ def get_success_url(self):
     def get_process_url(self):
         return reverse('process_payment', kwargs={'token': self.token})
 
+    def get_payment_url(self):
+        """
+        Get the url the view that handles the payment (payment_details() in documentation)
+        For now used only by PayU provider to redirect users back to CVV2 form
+        """
+        raise NotImplementedError()
+
+    def get_user(self):
+        """ Get the user asociated with this payment """
+        raise NotImplementedError()
+
+    def get_user_email(self):
+        """ Get user email """
+        try:
+            return self.get_user().email
+        except AttributeError:
+            return None
+
+    def get_user_first_name(self):
+        """
+        Get user first name
+        Used only by PayU provider for now
+        """
+        try:
+            return self.get_user().first_name
+        except AttributeError:
+            return None
+
+    def get_user_last_name(self):
+        """
+        Get user last name
+        Used only by PayU provider for now
+        """
+        try:
+            return self.get_user().last_name
+        except AttributeError:
+            return None
```

Comment:
This ties the package to a particular shape of the user object, it would not work in at least some of our projects. Could those fields not be extracted from billing information?

Reply:
These are the implementations for the most common case with intention, that users will override it in any other cases.

The billing information is info about the merchant, not the user, which is required by PayU (or PayU will ask that information in next step, which degrades UX)

Reply:
But maybe it would be appropriate to throw `NotImplementedError` instead of returning None and write some note to the comments about thet.